app/src/main/java/com/xmartlabs/template/ui/common/BaseRecyclerViewAdapter.java

```diff
@@ -0,0 +1,204 @@
+package com.xmartlabs.template.ui.common;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.MainThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.xmartlabs.template.helper.CollectionHelper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/**
+ * A Base Recycler View Adapter with already implemented functions such as
+ * Setting, removing, adding, getting items count.
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+public abstract class BaseRecyclerViewAdapter<T, VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {
+  protected static final int ITEM_TYPE_DIVIDER = -1;
+  protected static final int ITEM_TYPE_ITEM = -2;
+
+  @Getter(AccessLevel.PROTECTED)
+  @NonNull
+  private List<? extends T> items = new ArrayList<>();
+
+  /**
+   * Used to inflate the view layout/elements.
```

Comment:
I sincerely don't like the "Used to" prefix. Maybe "Inflates the view layout passed by param."?